sklearn/random_projection.py

```diff
@@ -78,9 +78,9 @@ def johnson_lindenstrauss_min_dim(n_samples, *, eps=0.1):
 
     Parameters
     ----------
-    n_samples : int or numpy array of int greater than 0,
-        Number of samples. If an array is given, it will compute
-        a safe number of components array-wise.
+    n_samples : int or array-like of thereof
```

Comment:
We have not reached a consensus in this regard: https://github.com/scikit-learn/scikit-learn/pull/18243#discussion_r475250655